src/App.tsx

```diff
@@ -35,6 +55,21 @@ function App() {
     sort,
   });
 
+  useEffect(() => {
+    setShowFavorites(false);
+  }, [debouncedQuery]);
+
+  useEffect(() => {
+    const newParams = new URLSearchParams();
+
+    if (query) newParams.set('search', query);
+    if (sort.sortBy) newParams.set('sortBy', sort.sortBy);
+    if (sort.order) newParams.set('order', sort.order);
```

Comment:
```
    query && newParams.set('search', query);
    sort.sortBy && newParams.set('sortBy', sort.sortBy);
    sort.order && newParams.set('order', sort.order);
```

Reply:
```
Expected an assignment or function call and instead saw an expression
.eslint[@typescript-eslint/no-unused-expressions](https://typescript-eslint.io/rules/no-unused-expressions)
```

I'm encountering the error "Expected an assignment or function call and instead saw an expression" from ESLint. What could be causing this?

Reply:
Sorry ESLint is right, mybad, the && conditional is being flagged as an unused expression.

Just keep the ifs. I would add the {} anyways, it adds a few more codelines but prevents errors and improves readability

Reply:
Perfect. So I'll keep the if statements and add the {} for improved readability and error prevention. Thanks for the help!